pyincore/semanticservice.py

```diff
@@ -0,0 +1,167 @@
+            return http_response
+        except requests.exceptions.HTTPError:
+            logger.error(
+                "A HTTPError has occurred \n"
+                + "HTTP Status code: "
+                + str(http_response.status_code)
+                + "\n"
+                + "Error Message: "
+                + http_response.content.decode()
+            )
+            raise
+        except requests.exceptions.ConnectionError:
+            logger.error(
+                "ConnectionError: Failed to establish a connection with the server. "
+                "This might be due to a refused connection. "
+                "Please check that you are using the right URLs."
+            )
+            raise
+        except requests.exceptions.RequestException:
+            logger.error(
+                "RequestException: There was an exception while trying to handle your request. "
+                "Please go to the end of this message for more specific information about the exception."
+            )
+            raise
+
+    def get_all_semantic_types(
+        self,
+        hyperlink: bool = False,
+        order: str = "asc",
+        skip: int = 0,
+        limit: int = 50,
+        detail: bool = False,
+        save_json: bool = False,
+        json_path: str = "",
+        timeout: Tuple[int, int] = (30, 600),
+        **kwargs,
+    ) -> list:
+        """Get all semantic types.
+
+        Args:
+            hyperlink (bool): If True, return a list of hyperlinks to the semantic types.
+            order (str): Order of the semantic types. Can be 'asc' or 'desc'.
+            skip (int): Number of semantic types to skip.
+            limit (int): Number of semantic types to return.
+            detail (bool): If True, return detailed information about the semantic types.
+            save_json (bool): If True, save the json response to a file.
+            json_path (str): Path to save the json response.
+            timeout (tuple): Timeout for the request in seconds.
+                            First value is the connection timeout, second value is the read timeout.
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list: List of semantic types.
+
+        """
+
+        url = self.base_url
+        payload = {
+            "hyperlink": hyperlink,
+            "order": order,
+            "skip": skip,
+            "limit": limit,
+            "detail": detail,
+        }
+        response = self.client.get(
+            url, params=payload, timeout=timeout, **kwargs
+        )
+        data = self.return_http_response(response).json()
+
+        if save_json:
+            with open(json_path, "w") as f:
+                json.dump(data, f, indent=4)
+
+        return data
+
+    def get_semantic_type_by_name(
+        self,
+        type_name: str,
+        save_json: bool = False,
+        json_path: str = "",
+        timeout: Tuple[int, int] = (30, 600),
+        **kwargs,
+    ) -> list:
```

Comment:
to dict maybe?

Reply:
no, the response we get be it single or multi, it's always a list... which is why I just kept to list

Reply:
you are right. 
But we should change. IMO It's equivlent to /dfr3/{id} or /dataset/{id}; returning a list is a bit confusing :-(

Reply:
Yep, I agree.